identity_stardust/src/document/stardust_document_metadata.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use core::fmt::Debug;
+use core::fmt::Display;
+use core::fmt::Formatter;
+
+use identity_core::common::Object;
+use identity_core::common::Timestamp;
+use identity_core::convert::FmtJson;
+use serde::Deserialize;
+use serde::Serialize;
+
+/// Additional attributes related to a [`StardustDocument`][crate::StardustDocument].
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+pub struct StardustDocumentMetadata {
+  // TODO: store created in the immutable metadata, if possible?
```

Comment:
Good idea, I'll see about that in the client impl.

Reply:
It would be nice but I don't know what format the immutable metadata is meant to take for other purposes: they don't have a dedicated `created` timestamp as far as I know. It might be fine for Alias Outputs used only as DID Documents though.